rust_src/src/frames.rs

```diff
@@ -406,6 +408,28 @@ pub fn frame_bottom_divider_width(frame: LispFrameOrSelected) -> i32 {
     frame.bottom_divider_width
 }
 
+/// Return scroll bar width of FRAME in pixels.
+#[lisp_fn(min = "0")]
+pub fn frame_scroll_bar_width(frame: LispFrameOrSelected) -> i32 {
+    let frame: LispFrameRef = frame.into();
+    match frame.vertical_scroll_bar_type() {
+        vertical_scroll_bar_type::vertical_scroll_bar_left
+        | vertical_scroll_bar_type::vertical_scroll_bar_right => frame.config_scroll_bar_width,
+        _ => 0,
+    }
+}
+
+/// Return scroll bar height of FRAME in pixels.
+#[lisp_fn(min = "0")]
+pub fn frame_scroll_bar_height(frame: LispFrameOrSelected) -> i32 {
+    let frame: LispFrameRef = frame.into();
+    if frame.horizontal_scroll_bars() {
```

Comment:
The C code uses `FRAME_HAS_HORIZONTAL_SCROLL_BARS` to check if the window system is live or not. We should port that logic using the feature support then this function can be re-implemented simply.

Reply:
My recent change for LIBXML2 is a good guide for this. It is quite simple.